Gems/ScriptCanvas/Code/Include/ScriptCanvas/Serialization/GraphDataSerializer.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/Json/JsonSerialization.h>
+#include <ScriptCanvas/Core/SerializationListener.h>
+#include <ScriptCanvas/Core/GraphData.h>
+#include <ScriptCanvas/Serialization/GraphDataSerializer.h>
+
+using namespace ScriptCanvas;
+
+namespace GraphDataSerializerCpp
+{
+    void CollectNodes(const GraphData::NodeContainer& container, SerializationListeners& listeners)
+    {
+        for (auto& nodeEntity : container)
+        {
+            if (nodeEntity)
+            {
+                if (auto listener = azrtti_cast<SerializationListener*>(AZ::EntityUtils::FindFirstDerivedComponent<Node>(nodeEntity)))
+                {
+                    listeners.push_back(listener);
+                }
+            }
+        }
+    }
+}
+
+namespace AZ
+{
+    AZ_CLASS_ALLOCATOR_IMPL(GraphDataSerializer, SystemAllocator, 0);
+
+    JsonSerializationResult::Result GraphDataSerializer::Load
+        ( void* outputValue
+        , [[maybe_unused]] const Uuid& outputValueTypeId
+        , const rapidjson::Value& inputValue
+        , JsonDeserializerContext& context)
+    {
+        namespace JSR = JsonSerializationResult;
+
+        AZ_Assert(outputValueTypeId == azrtti_typeid<GraphData>()
+            , "RuntimeVariableSerializer Load against output typeID that was not GraphData");
+        AZ_Assert(outputValue, "RuntimeVariableSerializer Load against null output");
+        context.GetMetadata().Add(SerializationListeners());
```

Comment:
If `GraphDataSerializer`s can be nested, have you tested that nested ones continue to work as expected?
Also this can be simplified to `context.GetMetadata().Create<SerializationListeners>();`

Reply:
GraphDataSerializers cannot be nested.